wagtail/admin/views/generic/mixins.py

```diff
@@ -287,14 +287,19 @@ def workflow_tasks(self):
 
     def user_has_permission(self, permission):
         user = self.request.user
-        if user.is_superuser:
-            return True
```

Comment:
This was added in 4a404aee2e62faefdfb0b4401fc661d2eb545818 as part of "workflows for snippets". I had to follow the logic for `can_lock()` and `can_unlock()` in `PagePermissionTester` (which always allows superusers to lock and unlock) to ensure the behaviour is exactly the same for both pages and snippets.

While doing that, I noticed that all the methods in `PagePermissionTester` always return `True` for superusers, except for page-specific cases (e.g. page is root, `creatable_subpage_models()` doesn't allow it, etc.) which don't apply for snippets. 

Hence, instead of adding the `if user.is_superuser: return True` only when checking for `lock` and `unlock` permissions, I opted to put it at the very beginning here. Of course, this ended up causing #11707 for people who have customised `permission_policy` so that even superusers are not allowed to do the action.

---

We technically can do it a different way: inside `ModelPermissionPolicy`, make it so that `user_has_permission` return `True` if the action is `lock`/`unlock` and the user is superuser. However, I didn't want to muddy the `ModelPermissionPolicy` too much until we get the permissions system worked out...